earthquakes/notebooks/Earthquakes.ipynb

```diff
@@ -0,0 +1,655 @@
+    "data_dir = \"D:/KaggleData/earthquakes/\"\n",
+    "plt.style.use(\"dark_background\") # uncomment if using light jupyter theme"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Load the data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train = pd.read_csv(os.path.join(data_dir, \"train.csv\"),\n",
+    "                    dtype={\"acoustic_data\": np.int16, \"time_to_failure\": np.float64})\n",
+    "\n",
+    "train.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\"We have {:,} rows and {} columns.\".format(train.shape[0], train.shape[1]))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's save the column names as variables for convenience:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "xcol = \"acoustic_data\"\n",
+    "ycol = \"time_to_failure\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Test data has a different structure: there is a separate CSV for each segment. Let's look at a couple of them:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# take the segment ids from the sample submission file\n",
+    "sample_submission = pd.read_csv(os.path.join(data_dir, \"sample_submission.csv\"), index_col=\"seg_id\")\n",
+    "x_test = pd.DataFrame(columns=[xcol], dtype=np.float64, index=sample_submission.index)\n",
+    "\n",
+    "# load first 10 segments\n",
+    "for seg_id in x_test.index[0:10]:\n",
+    "    segment = pd.read_csv(os.path.join(data_dir, \"test\", seg_id + \".csv\"))\n",
+    "    print(\"shape: {}\".format(segment.shape))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Test data consists of segments of 150,000 observations each."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 1. EDA\n",
+    "\n",
+    "## 1.1. High level stats"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train.describe()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "segment.describe()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Data is centered around 4 or 5, but has big outlier values (high as well as low). Let's look at this closer."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\"Number of earthquakes is {}\".format(np.sum(train[ycol] == 0)))"
```

Comment:
An earthquake does not necessarily mean that time_to_failure is zero right, since it can happen between two intervals. Shouldn't it be when to time_to_failure is increasing over time? This can be calculated by np.sum(train[ycol] - train[ycol].shift(1) > 0), which gives 16 earthquakes.

Reply:
You're right, 16 is the right number. Will update.